conf: fix support for private key PEM files other than PKCS#8

mikewadsten opened the pull request:
The Mbed TLS bug being worked around
(https://github.com/Mbed-TLS/mbedtls/issues/3896)
affects _all_ PEM inputs.
`conf_set_pem` only applied the workaround to certificates and PKCS#8 encoded keys.

MathisMARION commented on common/commandline.c:
```diff
@@ -303,7 +303,9 @@ void conf_set_pem(const struct storage_parse_info *info, void *raw_dest, const v
-        strstr(dest_str, "-----BEGIN PRIVATE KEY-----"))
+        strstr(dest_str, "-----BEGIN PRIVATE KEY-----") ||
+        strstr(dest_str, "-----BEGIN EC PRIVATE KEY-----") ||
+        strstr(dest_str, "-----BEGIN RSA PRIVATE KEY-----"))
```
Wi-SUN only supports EC. RSA support is not necessary.

MathisMARION reviewed (commented):
Hello and thank you for your contribution!

Interestingly, this syntax is not included in the official PEM file format specification ([RFC 7468](https://datatracker.ietf.org/doc/html/rfc7468)). However most crypto library implement it so it makes sense to support it ([OpenSSL](https://github.com/openssl/openssl/blob/openssl-3.4.1/include/openssl/pem.h#L57), [MbedTLS](https://github.com/Mbed-TLS/mbedtls/blob/v3.6.2/library/pk_internal.h#L40), [GnuTLS](https://gitlab.com/gnutls/gnutls/-/blob/3.8.9/lib/x509/privkey.c?ref_type=tags#L530)).

mikewadsten committed:
conf: fix support for private key PEM files other than PKCS#8

The Mbed TLS bug being worked around
(https://github.com/Mbed-TLS/mbedtls/issues/3896)
affects _all_ PEM inputs.
`conf_set_pem` only applied the workaround to certificates and
PKCS#8 encoded keys.

`-----BEGIN PRIVATE KEY-----` is the PKCS#8 PEM header called out in
RFC 7468. While `-----BEGIN EC PRIVATE KEY-----` is not mentioned in
that RFC, RFC 5915 does describe it as a "popular format" for using
PEM encoding to store an EC key, and most crypto libraries implement it.

mikewadsten commented:
> Hello and thank you for your contribution!
> 
> Interestingly, this syntax is not included in the official PEM file format specification ([RFC 7468](https://datatracker.ietf.org/doc/html/rfc7468)). However most crypto library implement it so it makes sense to support it ([OpenSSL](https://github.com/openssl/openssl/blob/openssl-3.4.1/include/openssl/pem.h#L57), [MbedTLS](https://github.com/Mbed-TLS/mbedtls/blob/v3.6.2/library/pk_internal.h#L40), [GnuTLS](https://gitlab.com/gnutls/gnutls/-/blob/3.8.9/lib/x509/privkey.c?ref_type=tags#L530)).

RFC 7468 specifically is about textual encodings of, among other things, PKCS structures, so I'm not surprised that RFC doesn't touch on other content structures. `-----BEGIN EC PRIVATE KEY-----` is mentioned in [RFC 5915](https://datatracker.ietf.org/doc/html/rfc5915#section-4) as a "popular format" of using PEM encoding to store the key.

MathisMARION commented:
> > Hello and thank you for your contribution!
> > Interestingly, this syntax is not included in the official PEM file format specification ([RFC 7468](https://datatracker.ietf.org/doc/html/rfc7468)). However most crypto library implement it so it makes sense to support it ([OpenSSL](https://github.com/openssl/openssl/blob/openssl-3.4.1/include/openssl/pem.h#L57), [MbedTLS](https://github.com/Mbed-TLS/mbedtls/blob/v3.6.2/library/pk_internal.h#L40), [GnuTLS](https://gitlab.com/gnutls/gnutls/-/blob/3.8.9/lib/x509/privkey.c?ref_type=tags#L530)).
> 
> RFC 7468 specifically is about textual encodings of, among other things, PKCS structures, so I'm not surprised that RFC doesn't touch on other content structures. `-----BEGIN EC PRIVATE KEY-----` is mentioned in [RFC 5915](https://datatracker.ietf.org/doc/html/rfc5915#section-4) as a "popular format" of using PEM encoding to store the key.

Good to know! Thank you for providing this reference. Do you mind mentioning it in the commit log?

mikewadsten commented:
> Good to know! Thank you for providing this reference. Do you mind mentioning it in the commit log?

Done.

MathisMARION commented:
Great! I have applied this patch onto our private development repository. It will be released as part of v2.3.2, and I will close this PR when it is made public.
